xrpl4j-binary-codec/src/main/java/org/xrpl/xrpl4j/codec/binary/XrplBinaryCodec.java

```diff
@@ -152,7 +152,13 @@ public String encodeForSigningClaim(String json) throws JsonProcessingException
    * @return A {@link String} representing the decoded hex.
    */
   public String decode(String hex) {
-    return new BinaryParser(hex).readType(STObjectType.class)
+    String nonSignPrefixHex;
+    if (hex.startsWith(TRX_SIGNATURE_PREFIX) || hex.startsWith(TRX_MULTI_SIGNATURE_PREFIX)) {
+      nonSignPrefixHex = hex.substring(8);
```

Comment:
I know both of these prefixes are 8 characters of hex, but we should probably do `hex.substring(TRX_SIGNATURE_PREFIX.length()` just to be sure, and to make this more readable.

You'll need two `if` blocks to do this fyi